examples/pytorch/gcmc/model.py

```diff
@@ -392,12 +391,7 @@ def forward(self, ufeat, ifeat):
         """
         ufeat = self.dropout(ufeat)
         ifeat = self.dropout(ifeat)
-        basis_out = []
-        for i in range(self._num_basis):
-            ufeat_i = ufeat @ self.Ps[i]
-            out = th.einsum('ab,ab->a', ufeat_i, ifeat)
-            basis_out.append(out.unsqueeze(1))
-        out = th.cat(basis_out, dim=1)
+        out = th.einsum('ai,bij,aj->ab', ufeat, self.P, ifeat)
```

Comment:
Just tried the gcmc example on this branch, on ml-10m(default setting) using einsum runs out of GPU memory.

Reply:
What @BarclayII did is to parallelize the message passing of multiple basis. However, PyTorch CUDA kernels are executed asynchronously so I don't think it's necessary to do so. (Though it might be faster in some cases, but after we merge featgraph it will not be the case).

One interesting direction is: can we enrich featgraph to support sophisticated einsum on edges?

Reply:
Reverted.